ReactWindows/ReactNative.Net46/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -225,6 +236,8 @@ public async void multiMerge(string[][] keyValueArray, ICallback callback)
         [ReactMethod]
         public async void clear(ICallback callback)
         {
+            if (await IsCancelledOrDisposed()) return;
+
             await _mutex.WaitAsync().ConfigureAwait(false);
```

Comment:
What happens to these WaitAsync calls which are already in progress? We should pass the cancellation token to them to ensure that they are cancelled before we dispose the mutex.
I tried prototyping a race condition and looks like not cancelling the wait results in a the thread waiting there for ever.